### PR TITLE
New button using default template in kanban view.

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
#### Ticket Link
Fixes https://github.com/mattermost/focalboard/issues/3275

### CODE REVIEW

**Comment by wiggin77** [comment]
@Rajat-Dabade there is a compile error.

File: webapp/src/components/kanban/kanban.test.tsx [hunk 1]
```diff
@@ -88,6 +89,12 @@ describe('src/component/kanban/kanban', () => {
                 board_id_1: {userId: 'user_id_1', schemeAdmin: true},
             },
         },
+        views: {
+            views: {
+                boardView: activeView,
+            },
+            current: 'boardView',
+        },
```
Comment (sbishel) [inline webapp/src/components/kanban/kanban.test.tsx:97]: Add tests that will verify `addCardFromTemplate` gets called as appropriate. Similar to 
`test('return kanban and click on New', () => {`

File: webapp/src/components/kanban/kanbanCard.tsx [hunk 2]
```diff
@@ -115,6 +115,7 @@ const KanbanCard = (props: Props) => {
                                 false,
                                 'duplicate card',
                                 false,
+                                undefined,
```
Comment (sbishel) [inline webapp/src/components/kanban/kanbanCard.tsx:118]: With my suggested changes you may need to pass an empty object `{}` here.

File: webapp/src/octoClient.ts [hunk 3]
```diff
@@ -665,15 +665,20 @@ class OctoClient {
         return this.getJson<BoardsAndBlocks>(response, {} as BoardsAndBlocks)
     }
 
-    async duplicateBlock(boardID: string, blockID: string, asTemplate: boolean): Promise<Block[] | undefined> {
+    async duplicateBlock(boardID: string, blockID: string, asTemplate: boolean, propertiesThatMeetFilters?: Record<string, string>): Promise<Block[] | undefined> {
```
Comment (sbishel) [inline webapp/src/octoClient.ts:668]: These changes shouldn't be necessary.

**Review by sbishel — CHANGES_REQUESTED** [review]
Your webapp changes look good, however, I think you can/should handle this on the webapp side and revert all your server changes.

Eventually, we do want to handle these things server-side, this will help performance, but we want to move everything when we do that.

So to stay consistent with how everything else works, you can make a single line change in `mutator.ts` and remove all the server side changes. 

Let me know if you want to meet to discuss.

File: webapp/src/mutator.ts [hunk 4]
```diff
@@ -975,12 +975,13 @@ class Mutator {
         fromTemplate = false,
         description = 'duplicate card',
         asTemplate = false,
+        propertiesThatMeetFilters?: Record<string, string> | undefined,
```
Comment (sbishel) [inline webapp/src/mutator.ts:978]: Let's rename this to something like `propertyOverrides`. That name has context in `kanban.tsx`, but not here.

File: webapp/src/mutator.ts [hunk 5]
```diff
@@ -975,12 +975,13 @@ class Mutator {
...
         afterRedo?: (newCardId: string) => Promise<void>,
         beforeUndo?: () => Promise<void>,
     ): Promise<[Block[], string]> {
         return undoManager.perform(
             async () => {
-                const blocks = await octoClient.duplicateBlock(boardId, cardId, asTemplate)
+                const blocks = await octoClient.duplicateBlock(boardId, cardId, asTemplate, propertiesThatMeetFilters)
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply webapp/src/mutator.ts]: Got it.

File: webapp/src/components/kanban/kanban.test.tsx (hunk 1 above)
Comment (Rajat-Dabade) [reply webapp/src/components/kanban/kanban.test.tsx:96]: Sure

File: webapp/src/octoClient.ts (hunk 3 above)
Comment (Rajat-Dabade) [reply webapp/src/octoClient.ts]: Yes, got it.

File: webapp/src/mutator.ts (hunk 4 above)
Comment (Rajat-Dabade) [reply webapp/src/mutator.ts]: Done.

File: webapp/src/components/kanban/kanbanCard.tsx (hunk 2 above)
Comment (Rajat-Dabade) [reply webapp/src/components/kanban/kanbanCard.tsx]: Done

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Pull request merged**